extension/persistence/eclipselink/src/main/java/io/polaris/extension/persistence/impl/eclipselink/PolarisEclipseLinkStore.java

```diff
@@ -409,4 +409,33 @@ void deletePrincipalSecrets(EntityManager session, String clientId) {
 
     session.remove(modelPrincipalSecrets);
   }
+
+  boolean locationOverlapsWithExistingTableLike(EntityManager session, String location) {
+    if (location == null) {
+      return false;
+    }
+    diagnosticServices.check(session != null, "session_is_null");
+
+    return session
+        .createQuery(
+            """
+                SELECT
+                    1
+                FROM
+                    ModelEntityActive
+                WHERE
+                    location IS NOT NULL
+                    AND typeCode = :table_code
+                    AND (
+                        location LIKE CONCAT(:location, '%')
+                        OR :location LIKE CONCAT(location, '%')
```

Comment:
Doesn't the 2nd condition cause a full-table-scan?

Reply:
Yes, as written it unfortunately will.

Of course, this is all dependent on the metastore manager implementation (and in this case the backing database's implementation).

In any event if this check was _always_ being done, I think there are some easy ways to optimize this. But since the check is optional, we may need to check every path. I'm still testing ways to improve performance of this optional check.

Reply:
I was thinking about this problem a bit. It's a tricky-fun one ;)
Just brain-dumping some thoughts:

There are a few things that play a role here - respectively w/ locations in general.

I _think_ we can rely on `/` as a separator. With that in mind, a location is a duple of "bucket" (S3/GCS bucket or ADSL fs-name) plus a list of path **elements**. If we distinguish _parent directories_ from _table directories_, the check becomes easier. I.e. a separate metastore entity that is only used to track locations.
```sql
CREATE TABLE locations (
  bucket TEXT NOT NULL,  -- storage bucket, e.g. s3://bucket/
  path TEXT NOT NULL,   -- storage location path, e.g. my/path/to/my-table
  kind TEXT NOT NULL,    -- marker for "parent-directory" or "table-location"
  entity_id TEXT NULL   -- id of the table
);
```
When you want to add/check for a new `location` like `s3://bucket/my/path/my-table`, the following `INSERT`s could do the trick:
```sql
INSERT INTO locations (bucket, path, kind, entity_id) VALUES ( 's3://bucket', `my`, `parent`, NULL) ON CONFLICT DO NOTHING;
INSERT INTO locations (bucket, path, kind, entity_id) VALUES ( 's3://bucket', `my/path`, `parent`, NULL) ON CONFLICT DO NOTHING;
INSERT INTO locations (bucket, path, kind, entity_id) VALUES ( 's3://bucket', `my/my-table`, `table`, '1234');
```
If the last one fails -> location already used -> fail hard. If any of the parents did not succeed, verify that those are all `kind = 'parent'`.

I suspect, this needs some more thought around race conditions (two tables with conflicting locations).